shared/constants/metametrics.js

```diff
@@ -0,0 +1,155 @@
+ * necessary.
+ * @typedef {Object} MetaMetricsContext
+ * @property {Object} app
+ * @property {string} app.name - the name of the application tracking the event
+ * @property {string} app.version - the version of the application
+ * @property {string} userAgent - the useragent string of the user
+ * @property {MetaMetricsPageObject} [page]     - an object representing details
+ *  of the current page
+ * @property {Object} [referrer] - for metamask, this is the dapp that triggered
+ *  an interaction
+ * @property {string} [referrer.url] - the origin of the dapp issuing the
+ *  notification
+ */
+
+/**
+ * @typedef {Object} MetaMetricsEventPayload
+ * @property {string}  event - event name to track
+ * @property {string}  category - category to associate event to
+ * @property {string} [environmentType] - The type of environment this event
+ *  occurred in. Defaults to the background process type
+ * @property {object}  [properties] - object of custom values to track, keys
+ *  in this object must be in snake_case
+ * @property {object}  [sensitiveProperties] - Object of sensitive values to
+ *  track. Keys in this object must be in snake_case. These properties will be
+ *  sent in an additional event that excludes the user's metaMetricsId
+ * @property {number}  [revenue] - amount of currency that event creates in
+ *  revenue for MetaMask
+ * @property {string}  [currency] - ISO 4127 format currency for events with
+ *  revenue, defaults to US dollars
+ * @property {number}  [value] - Abstract business "value" attributable to
+ *  customers who trigger this event
+ * @property {MetaMetricsDynamicContext} [eventContext] - additional context
+ *  to attach to event
+ */
+
+/**
+ * @typedef {Object} MetaMetricsEventOptions
+ * @property {boolean} [isOptIn] - happened during opt in/out workflow
+ * @property {boolean} [flushImmediately] - When true will automatically flush
+ *  the segment queue after tracking the event. Recommended if the result of
+ *  tracking the event must be known before UI transition or update
+ * @property {boolean} [excludeMetaMetricsId] - whether to exclude the user's
+ *  metametrics id for anonymity
+ * @property {string}  [metaMetricsId] - an override for the metaMetricsId in
+ *  the event one is created as part of an asynchronous workflow, such as
+ *  awaiting the result of the metametrics opt-in function that generates the
+ *  user's metametrics id
+ * @property {boolean} [matomoEvent] - is this event a holdover from matomo
+ *  that needs further migration? when true, sends the data to a special
+ *  segment source that marks the event data as not conforming to our schema
+ * @property {number} metaMetricsSendCount - the count of the number of sends
+ *  that have triggered a metrics event used to determine if we should allow
+ *  the user's id to be included in the payload periodically
+ */
+
+export const METAMETRICS_ANONYMOUS_ID = '0x0000000000000000'
+
+/**
+ * This object is used
```

Comment:
it is, i swear. I'll work on a few documentation things I've already found -- but should still be good for code review.